GUIDELINES.md

```diff
@@ -0,0 +1,139 @@
+    * Example: `fun givenLoginException { 'when'(instance.loginOrRegister(anyString(), anyString())).thenThrow(AuthManager.WrongPasswordException()) }`
+  * Configure dependencies
+    * You have already configured dependencies on a module level now, but you need to configure dependencies also on a steps (or in some rare cases on test) level too
+    * You can start configuring in a steps or test class with writing  `override fun configure() = super.configure()` and then you can add...
+    * Example: `.onSetUp { ... }` (runs the given block before any test is run)
+    * Example: `.requireMock<BackendGateway>()` (tells the dependency management that this steps class is requiring or acting on a mock of the type `BackendGateway`)
+    * Example: `.requireReal<LoginViewModel>()` (the same for real)
+    * Example: `.offerMock<BackendGateway> { FakeBackendGateway() }` (provides an implementation for a `BackendGateway` to dependency management, so if anyone in the test system needs a mock of a `BackendGateway` the `FakeBackendGateway` implementation will be initialized)
+    * Example: `.offerMockRequired<BackendGateway> { FakeBackendGateway() }` (on top on the former it also forces dependency management that `BackendGateway` can't be configured to be real)
+    * Example: `.offerReal<LoginViewModel> { LoginViewModelImpl(instanceOf(), someOtherValue) }` (provides an implementation for a `LoginViewModel` to dependency management, where in this example it's demonstrated how `instanceOf()` can be used to automatically inject the correct type into the constructor as well as some other custom value)
+    * Example: `.offerRealRequired<LoginViewModel> { LoginViewModelImpl(instanceOf(), someOtherValue) }` (on top on the former it also forces dependency management that `LoginViewModel` can't be configured to be mock)
+    * As you can see, you can control default configuration and initialization behavior on a module basis _and_ on a steps basis
+  * The easiest way to approach configuration is to just allow what's necessary on a module level and make it more specific on the test or steps level
+    * E.g. if there is a `AuthManager` which can be used as a mock and a real instance as well, configure it as `any` in module configuration
+    * If in a test system there is an integration setup which requires `AuthManager` to be mock and a different setup which required `AuthManager` to be a real instance, consider the following:
+    * Create a steps class which is specialized on mocking `AuthManager` and call it `AuthManagerMockSteps`
+      * there you also configure the `AuthManager` dependency to be a mock only by using `requireMockOnly<>`
+* If you have done that we should already have a structure like this
+  * If possible, the test class refers to a **user-facing feature** under test (e.g. "LoginTest", "TemperatureDisplayTest", "DataLoggingTest", ...)
+  * The the test functions are named after scenarios
```

Comment:
All of the above is very detailed 👍 
However I would much rather have some short examples at the top for a quick overview.
As someone who codes but has never seen this library in action before or not worked with it, I think such an example could give a very quick grasp:

``` kotlin
/**
 * Explanation
 */
val appModuleTestingConfiguration = moduleTestingConfiguration(
    dependentConfig1 // explanation
) {
    dependency any of<Whatever>()  // explanation
    dependency mockOnly of<Whatever2>()  // explanation
    ....
}

// needs to be tested
class Someclass @Inject constructor(val whatever: Whatever) { ... }

/**
 * Explanation
 */
class SomeClassTest : BaseJUnitTest(appModuleTestingConfiguration) {
    private val sut by steps<SomeClassSteps>()

    @Test
    fun `bla blue bli`() = sut {
        given()
        when()
        then()
    }
}

/**
 * Explanation
 */
class SomeClassSteps : (
    testContext: TestContext // explanation
) : BaseSteps(testContext, appModuleTestingConfiguration) {
    ....
}
```